enclave.go

```diff
@@ -269,14 +306,21 @@ func (e *Enclave) Stop() error {
 	if err := e.pubSrv.Shutdown(context.Background()); err != nil {
 		return err
 	}
+	if err := e.promSrv.Shutdown(context.Background()); err != nil {
+		return err
+	}
 	return nil
 }
 
-// startWebServers starts both our public-facing and our enclave-internal Web
-// server in a goroutine.
+// startWebServers starts our public-facing Web server, our enclave-internal
+// Web server, and -- if desired -- a Web server for profiling and/or metrics.
 func startWebServers(e *Enclave) error {
-	elog.Printf("Starting public (%s) and private (%s) Web server.", e.pubSrv.Addr, e.privSrv.Addr)
+	if e.cfg.PrometheusPort > 0 {
+		elog.Printf("Starting Prometheus Web server (%s).", e.promSrv.Addr)
+		go e.promSrv.ListenAndServe() //nolint:errcheck
```

Comment:
This should be `ListenAndServeTLS` I think, since it's facing outside the enclave. Even if we don't publicly  route the port, we should encrypt transfer between machines.

Reply:
Addressed in feaf133483114ec649785bc4e3b6d41173f963f6.